test/library/standard/Sys/sockaddr.chpl

```diff
@@ -0,0 +1,78 @@
+use Sys;
+use CPtr;
+use SysCTypes;
+use UnitTest;
+
+proc test_addr_string_ipv4(test: borrowed Test) throws {
+  var socket_addr:sys_sockaddr_t = new sys_sockaddr_t();
+  var address = "127.0.0.1";
+  var port:uint(16) = 8000;
+  var family = AF_INET;
+  socket_addr.initialize(address.c_str(), port, family);
+
+  var familySize = INET_ADDRSTRLEN;
+  var buffer = c_calloc(c_char,familySize);
+  var getPort:uint(16);
+  sys_extract_sys_sockaddr_t(socket_addr,buffer,getPort);
+  var getAddr = createStringWithOwnedBuffer(buffer,familySize,familySize);
+
+  test.assertTrue(getAddr.c_str() == address.c_str());
```

Comment:
here it's failing on direct comparison. I am not able to figure out why is it so

Reply:
I'd recommend running the program under `valgrind` to look for use-after-free type errors. Does the situation change if you double the allocation size? What if you check `getAddr == address` ?

Reply:
the issue was related to the call in `createStringfromBorrowed` where I wasn't specifying length properly so I changed the function to use `strlen` and return the use it in the call.